backends/tfhe-cuda-backend/cuda/src/integer/scalar_comparison.cuh

```diff
@@ -304,4 +304,116 @@ __host__ void host_integer_radix_scalar_maxmin_kb(
       stream, lwe_array_out, mem_ptr->tmp_lwe_array_out, lwe_array_left,
       lwe_array_right, mem_ptr->cmux_buffer, bsk, ksk, total_num_radix_blocks);
 }
+
+template <typename Torus>
+__host__ void host_integer_radix_scalar_equality_check_kb(
+    cuda_stream_t *stream, Torus *lwe_array_out, Torus *lwe_array_in,
+    Torus *scalar_blocks, int_comparison_buffer<Torus> *mem_ptr, void *bsk,
+    Torus *ksk, uint32_t num_radix_blocks, uint32_t num_scalar_blocks) {
+
+  auto params = mem_ptr->params;
+  auto big_lwe_dimension = params.big_lwe_dimension;
+  auto message_modulus = params.message_modulus;
+
+  auto eq_buffer = mem_ptr->eq_buffer;
+
+  size_t big_lwe_size = big_lwe_dimension + 1;
+  size_t big_lwe_size_bytes = big_lwe_size * sizeof(Torus);
+
+  auto scalar_comparison_luts = eq_buffer->scalar_comparison_luts;
+
+  uint32_t num_halved_scalar_blocks =
+      (num_scalar_blocks / 2) + (num_scalar_blocks % 2);
+
+  uint32_t num_lsb_radix_blocks =
+      std::min(num_radix_blocks, 2 * num_halved_scalar_blocks);
+  uint32_t num_msb_radix_blocks = num_radix_blocks - num_lsb_radix_blocks;
+  uint32_t num_halved_lsb_radix_blocks =
+      (num_lsb_radix_blocks / 2) + (num_lsb_radix_blocks % 2);
+
+  auto lsb = lwe_array_in;
+  auto msb = lwe_array_in + big_lwe_size * num_lsb_radix_blocks;
+
+  auto lwe_array_lsb_out = mem_ptr->tmp_lwe_array_out;
+  auto lwe_array_msb_out =
+      lwe_array_lsb_out + big_lwe_size * num_halved_lsb_radix_blocks;
+
+  cuda_synchronize_stream(stream);
```

Comment:
After this line we have two concurrent CPU threads. Each one will run on their own streams, `lsb_stream` and `msb_stream`.  I need to synchronize the main stream here to be sure all previous computation are completed before those streams start, otherwise some of them may start doing something on data that are still being computed in the main stream and we will have race conditions.